packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -677,43 +735,29 @@ export class IntervalCollectionView<TInterval extends ISerializableInterval> ext
         intervalType: MergeTree.IntervalType,
         props?: MergeTree.PropertySet,
     ) {
-        let seq = 0;
-        if (this.client) {
-            seq = this.client.getCurrentSeq();
-        }
+        const interval: TInterval = this.localCollection.addInterval(start, end, intervalType, props);
 
-        const serializedInterval: ISerializedInterval = {
-            end,
-            intervalType,
-            properties: props,
-            sequenceNumber: seq,
-            start,
-        };
-
-        return this.addInternal(serializedInterval, true, undefined);
-    }
-
-    public delete(
-        start: number,
-        end: number) {
-        let sequenceNumber = 0;
-        if (this.client) {
-            sequenceNumber = this.client.getCurrentSeq();
+        if (interval) {
+            const serializedInterval = {
+                end,
+                intervalType,
+                properties: interval.properties,
+                sequenceNumber: this.client?.getCurrentSeq() ?? 0,
+                start,
+            };
+            // Local ops get submitted to the server. Remote ops have the deserializer run.
+            this.emitter.emit("add", undefined, serializedInterval);
         }
 
-        const serializedInterval: ISerializedInterval = {
-            start,
-            end,
-            sequenceNumber,
-            intervalType: MergeTree.IntervalType.Transient,
-        };
+        this.emit("addInterval", interval, true, undefined);
 
-        this.deleteInterval(serializedInterval, true, undefined);
+        return interval;
     }
 
     // TODO: error cases
-    public addInternal(serializedInterval: ISerializedInterval, local: boolean, op: ISequencedDocumentMessage) {
-        const interval = this.localCollection.addInterval(
+    public addInternal(
```

Comment:
>addInternal [](http://example.com/codeflow?start=11&length=11)

should the above add method call this

Reply:
looks like you moved some methods which make the merge gross, but i'm not sure why anything in add needed to change

Reply:
add used to call addInternal, which would hit the case below

Reply:
I changed it to make the local and non-local cases more distinct. I had to do that for delete. It does make debugging easier if addInternal is only called in the case where an op comes over the wire. Sorry about the merge, though.